docs/source/python/api/compute.rst

```diff
@@ -504,6 +505,7 @@ Compute Options
    IndexOptions
    JoinOptions
    MakeStructOptions
+   MapArrayLookupOptions
```

Comment:
Just a reminder to revisit and implement this. 

To do so, you'll need to:

1. Add the options class definition to the Cython pxd: https://github.com/apache/arrow/blob/39367db2dab321dbbf4d12d2229020614b049dde/python/pyarrow/includes/libarrow.pxd#L2147-L2150
2. Add a Python options class wrapper to the Cython code: https://github.com/apache/arrow/blob/39367db2dab321dbbf4d12d2229020614b049dde/python/pyarrow/_compute.pyx#L1317-L1333
3. Add an import so it's in the public API: https://github.com/apache/arrow/blob/39367db2dab321dbbf4d12d2229020614b049dde/python/pyarrow/compute.py#L18
4. Add the options class to the serialization test: https://github.com/apache/arrow/blob/39367db2dab321dbbf4d12d2229020614b049dde/python/pyarrow/tests/test_compute.py#L138 (just add it to the list there, no need to add additional assertions to the test)
5. Add a brief test to make sure everything works (no need to add detailed tests, we just want to make sure the bindings were set up): https://github.com/apache/arrow/blob/39367db2dab321dbbf4d12d2229020614b049dde/python/pyarrow/tests/test_compute.py#L2081-L2092